### PR TITLE
Added etag handling

### PR DESCRIPTION
**Pull request by mitsuhiko**
Replaces #1217

### CODE REVIEW

**Comment by mitsuhiko** [comment]
This just keeps the changes small. The codestyle is pretty ugly in Werkzeug ATM and i will do a separate change to reindent it somewhat.

File: werkzeug/http.py [hunk 1]
```diff
@@ -899,6 +899,10 @@ def is_resource_modified(environ, etag=None, data=None, last_modified=None,
                 # entity-tags for If-None-Match"
                 unmodified = if_none_match.contains_weak(etag)
 
+            if_match = parse_etags(environ.get('HTTP_IF_MATCH'))
+            if if_match:
+                unmodified = not if_match.is_strong(etag)
```
Comment (riv2q) [inline werkzeug/http.py]: I think is worth to have this notation here
```
# https://tools.ietf.org/html/rfc7232#section-3.1
# "Origin server MUST use the strong comparison function when
# comparing entity-tags for If-Match"
```
same reasons as for  ```If-None-Match``` case

**Comment by riv2q** [comment]
@mitsuhiko Thanks for the interest around this pull-request.

File: werkzeug/http.py (hunk 1 above)
Comment (mitsuhiko) [reply werkzeug/http.py]: Indeed. Added that back.

**Comment by mitsuhiko** [comment]
@riv2q thanks for contributing this.

**Pull request merged**